implementations/rust/ockam/ockam_identity/src/models/credential_and_purpose_key.rs

```diff
@@ -13,3 +20,82 @@ pub struct CredentialAndPurposeKey {
     /// [`Credential`] and will be used to verify it
     #[n(2)] pub purpose_key_attestation: PurposeKeyAttestation,
 }
+
+impl CredentialAndPurposeKey {
+    /// Encode the credential as a hex String
+    pub fn encode_as_string(&self) -> Result<String> {
```

Comment:
I wonder if we can auto-implement these 4 functions for all types implementing `Encode` and `Decode`? Also, I would consider more specific functions name, something like `encode_as_cbor_bytes`

Reply:
P.S. This is a very good change

Reply:
> I wonder if we can auto-implement these 4 functions for all types implementing Encode and Decode? Also, I would consider more specific functions name, something like encode_as_cbor_bytes

Yes that could be some useful extension methods. Let's do this in another PR.